packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart

```diff
@@ -299,6 +299,99 @@ flutter:
     },
   );
 
+  testUsingContext(
+    'dart format is run when --format-arg is passed',
+    () async {
+      final File arbFile = fileSystem.file(fileSystem.path.join('lib', 'l10n', 'app_en.arb'))
+        ..createSync(recursive: true);
+      arbFile.writeAsStringSync('''
+{
+  "helloWorld": "Hello, World!",
+  "@helloWorld": {
+    "description": "Sample description"
+  }
+}''');
+      final File pubspecFile = fileSystem.file('pubspec.yaml')..createSync();
+      pubspecFile.writeAsStringSync(BasicProjectWithFlutterGen().pubspec);
+      processManager.addCommand(
+        const FakeCommand(
+          command: <String>[
+            'Artifact.engineDartBinary',
+            'format',
+            '/lib/l10n/app_localizations_en.dart',
+            '/lib/l10n/app_localizations.dart',
+            '--page-width=120',
+            '--enable-experiment',
+          ],
+        ),
+      );
+
+      final GenerateLocalizationsCommand command = GenerateLocalizationsCommand(
+        fileSystem: fileSystem,
+        logger: logger,
+        artifacts: artifacts,
+        processManager: processManager,
+      );
+
+      await createTestCommandRunner(command).run(<String>[
+        'gen-l10n',
+        '--format',
+        '--format-arg',
+        '--page-width=120',
+        '--format-arg',
+        '--enable-experiment',
+      ]);
+
+      final Directory outputDirectory = fileSystem.directory(fileSystem.path.join('lib', 'l10n'));
```

Comment:
Can you replace the test cases which check for files with the following pattern:

```dart
// `exists` is part of package:file_testing
expect(fileSystemEntity, exists);
```